package.json

```diff
@@ -16,24 +16,17 @@
         "LICENSE",
         "README.md"
     ],
-    "type": "module",
     "sideEffects": false,
-    "main": "lib/index.cjs.js",
-    "module": "lib/index.esm.js",
-    "esnext": "lib/index.js",
-    "types": "lib/index.d.ts",
-    "browser": {
-        "./lib/index.cjs.js": "./lib/index.browser.cjs.js",
-        "./lib/index.esm.js": "./lib/index.browser.esm.js"
+    "main": "lib/cjs/index.js",
+    "exports": {
+        "import": "./lib/esm/index.js",
+        "require": "./lib/cjs/index.js"
     },
-    "browserslist": [
-        "defaults",
-        "not IE 11",
-        "maintained node versions"
-    ],
+    "types": "lib/esm/index.d.ts",
     "scripts": {
         "clean": "shx rm -rf lib",
-        "build": "yarn clean && tsc && rollup -c",
+        "build": "yarn clean && tsc -p tsconfig.json && tsc -p tsconfig-cjs.json",
+        "postbuild": "echo '{\"type\":\"commonjs\"}' > lib/cjs/package.json && echo '{\"type\":\"module\"}' > lib/esm/package.json",
```

Comment:
this was the lesser of the two evils. Cleaner than changing all the extensions in the `lib/cjs` directory